src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/TypeBuilderImpl.cs

```diff
@@ -73,14 +75,24 @@ protected override void AddInterfaceImplementationCore([DynamicallyAccessedMembe
             _interfaces.Add(interfaceType);
         }
 
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2083:DynamicallyAccessedMembers", Justification = "Not sure how to handle")]
         [return: DynamicallyAccessedMembers((DynamicallyAccessedMemberTypes)(-1))]
-        protected override TypeInfo CreateTypeInfoCore() => throw new NotImplementedException();
+        protected override TypeInfo CreateTypeInfoCore() => this;
         protected override ConstructorBuilder DefineConstructorCore(MethodAttributes attributes, CallingConventions callingConvention, Type[]? parameterTypes, Type[][]? requiredCustomModifiers, Type[][]? optionalCustomModifiers) => throw new NotImplementedException();
         protected override ConstructorBuilder DefineDefaultConstructorCore(MethodAttributes attributes) => throw new NotImplementedException();
         protected override EventBuilder DefineEventCore(string name, EventAttributes attributes, Type eventtype) => throw new NotImplementedException();
 
         protected override FieldBuilder DefineFieldCore(string fieldName, Type type, Type[]? requiredCustomModifiers, Type[]? optionalCustomModifiers, FieldAttributes attributes)
         {
+            if (_enumUnderlyingType == null && IsEnum)
+            {
+                if ((attributes & FieldAttributes.Static) == 0)
```

Comment:
Why the check for Static? Was this a bug prior?

Reply:
Not it is checking if this is the underlying field, the underlying field is not static
https://github.com/dotnet/runtime/blob/d6a40b1874cac586fe28caf22843ce2324a9cae3/src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/EnumBuilderImpl.cs#L21-L22
Where fields for enum values literal are defined as static https://github.com/dotnet/runtime/blob/d6a40b1874cac586fe28caf22843ce2324a9cae3/src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/EnumBuilderImpl.cs#L30-L35